conductor.py

```diff
@@ -985,6 +689,99 @@ def log_time():
        states_file.write('\n'+str(tuple((datetime.now().isoformat(), j))))
    return ''
 
+# JSON file upload function to allow first .json upload and check text format 
+# It checks "id:" "title:" and "candidates:"
+@app.route('/upload-ballot-manifest-json', methods=['POST'])
+def upload_json():
+    # "Be conservative in what you send, be liberal in what you accept"
+    # TODO: for transparency, also return the file's hash
+    if 'file' not in request.files:
+        return 'File not uploaded', 400
+    else:
+        print("File found")
+    file = request.files['file']
+    contest_name = request.form['contest_name']
+    # "if user does not select file, browser also"
+    # "submit an empty part without filename"
+    if file.filename == '':
+        return 'No selected file', 400
+    if file:  # and allowed_file(file.filename):
+        filename = os.path.join(app.config['UPLOAD_FOLDER'], secure_filename(file.filename))
+        file.save(filename)
+        with open(filename, 'r') as FO:
+            uploaded_content = FO.read()
+        try:
+            global all_contests_bristol
+            # reading it as json
+            uploaded_json = json.loads(json.dumps(ast.literal_eval(uploaded_content)))
+            # clearing all_contests_bristol
+            all_contests_bristol = []
+            for current_dic in uploaded_json:
+                # making list and dictionaries from json
+                item_dic = {'id': current_dic['id'], 'title': current_dic['title'], 'candidates': []}
+                for candidate_item in current_dic['candidates']:
+                    item_dic['candidates'].append(candidate_item)
+                all_contests_bristol.append(item_dic)
+
+            global default_audit_state
+            # updating default_audit_state key that depend on all_contests_bristol
+            default_audit_state['all_contests']['ballot_comparison'] = all_contests_bristol
+
+            global audit_types
+            # updating audit_types key that depend on all_contests_bristol
+            audit_types['ballot_comparison']['all_contests'] = all_contests_bristol
+
+            global audit_state
+            audit_state = copy.deepcopy(default_audit_state)
```

Comment:
Why are we resetting to `default_audit_state` here? That'll reset all of our state; do we want that?